### DISCUSSION TITLE
Feature request

### DISCUSSION THREAD

**Discussion by 7enz** [helgoboss/helgobox#745] [Ideas]
FEATURE REQUEST:  monitoring GUI showing plugin parameter names/sliders

having a monitoring GUI interface that would do the following with Realearn
Gui interface=Parameter name/current value(slider)

a GUI monitoring interface that allows for X amount of columns and X amount of rows.. which is user set
examples 1x32 2x32 2x16 8x8 6x4 8x2 etc user definable
reason for the adjustable columns and rows, is screen realestate and what a user wishes to keep monitoring when also adjusting parameters
so a user might tailor the Gui to their given project template...

spacers or visual line-markers on the GUI(user adjustable)
if for example a column of 1x32 parameter name& sliders were to be the GUI... having a visual line(spacer) to distinguish where every 8th or 6th parameter ends & starts  would help with seeing relative positions with respect to encoders becos of varying controllers layouts

non focussed plugin-parameter names to appear in a GUI preset list (selection based -: xnumber of parameters from vst1  xnumber of parameters from vst2 xnumber parameters from vst3 etc)
user to create order of parameters and which FXnames show up in GUI

sliders to merely show current state of parameters and provide feedback(where possible)
if the parameter might be a switch, a simple "/" to denote it is button action

Quick learn feature as per Lbx smart knobs script would be great too
cheeRs
Ps: i tried to keep it simple

**Comment by helgoboss** [comment]
Main question: Are you only interested in setting up FX parameters with this? Or all kinds of things, e.g. track volume, all targets basically.

Also, I didn't get the part about "non focussed plugin-parameter names to appear in a GUI preset list".

**Reply by 7enz** [reply]
it would be short sighted of me to have it simply limited to just FX, to have the choices to add more targets... volumes, pans, parent volumes etc etc im sure would be extremely useful!!

non focussed plugin- parameter names was a reference to, Fx plugin's that doesnt  require track selection or focus... but im thinking this may again be short sighted?? why limit it? The idea behind it, was to have any parameter from any captured Vst(in realearn already or newly captured)... available to be put on the Gui for either monitoring or adjustment

**Reply by 7enz** [reply]
another addition that i think will be helpful is have a user area input 2 or 4 character area for user input as a mental reference they might use for their controller.. which is saved alongside the preset(having no function only used as reference)
example: where xxxx is used for mental reference

[ xxxx ] [target-name] [slider position]  [slidervalue]

**Reply by 7enz** [reply]
another addition LOL(dont want to complicate things)
but perhaps a states section... ie if a user wanted to monitor a particular state of lets say Armed tracks or Polled elements
mainly to do with button states i guess
perhaps using a numbering system to show states [01] for on and [00]
or perhaps if a button implements incremental buttons which uses a value sequence(glue section) to represent the current state as numbers again 0 being the least and the last incremental number being the highest number,
if value sequence has 8 states, and it is currently on the 4th value its represented as [04]

perhaps the interface could be something along the lines of..
[group type] [y]..
[xxxx] [target-name] [slider position][slider value]

where the top line is a header & y is based on the number of targets below the header
so if there are 7 targets below that have buttons [y] would appear 7 times..

**Reply by 7enz** [reply]
if it were also possible... to consider also having a last touched target value on a per [group type] & per [target-name] basis

perhaps needing only last 2 last affected [target-name] per group..
[group type] [y]
[xxxx] [target-name] [z] [slider position][slider value]

where [z] is last touched either a 1 or 2 upon touching... 2 if a target within that group is also touched.. alternating between 1&2

so a numbering system again might be used across all group types
where each last touched target value would go to 1 and every other target across a group and within a group is incremented by 1 
....but number only incremented across groups ..if a change takes place across a different [group type]
if in group type 1 a change occurs - this will only increment last touched target in another group by 1 or the maximum 2 if 2 targets are changed in the same group type
this just provides a mental method of backtracking on changes that have occurred

hopefully that makes sense lol

**Reply by 7enz** [reply]
another addition... 
things might be going overboard with this addition feature request
if the monitoring aspect, should change and morph into a control mechanism as well...
perhaps having a Master link(control)... so that if we want to control 4 parameters.. we can assign a master(button or encoder) to control other parameters without manually changing their original source.
